superdesk/io/feeding_services/ftp.py

```diff
@@ -69,74 +71,120 @@ def _test(self, provider):
         except Exception as ex:
             raise IngestFtpError.ftpError(ex, provider)
 
+    def _move(self, ftp, src, dest):
+        """Move distant file
+
+        :param ftp: FTP instance to use
+        :type ftp: ftplib.FTP
+        :param src: source path of the file to move
+        :type src: str
+        :param dest: dest path of the file to move
+        :type dest: str
+        """
+        try:
+            ftp.rename(src, dest)
+        except ftplib.all_errors as e:
+            logger.warning("Can't move file from {src} to {dest}: {reason}".format(
+                src=src,
+                dest=dest,
+                reason=e))
+
+    def _create_if_missing(self, ftp, path):
+        """Check if a dir exists, and create it else
+
+        :param ftp: FTP instance to use
+        :type ftp: ftplib.FTP
+        :param src: dir path to check
+        """
+        try:
+            next(ftp.mlsd(path))
```

Comment:
I'm not sure if it's the best way to check it, when there are lots of files it could take some time, or even timeout.
What about trying to create that dir and ignore any conflict error?